sql/dd/dd_table.cc

```diff
@@ -2190,6 +2190,7 @@ static std::unique_ptr<dd::Table> create_dd_system_table(
                           ? dd::Abstract_table::HT_HIDDEN_SYSTEM
                           : dd::Abstract_table::HT_VISIBLE);
 
+  file->adjust_create_info_for_dd(create_info);
```

Comment:
I think, with global DD, this is not the right approach. dd::Table can be adjusted by InnoDB at create_table API.

And remove this API.